WindowsBuilder.ps1

```diff
@@ -0,0 +1,250 @@
+
+	Write-Yellow "Downloading libuv..."
+	git clone "$url_libuv"
+	Write-Yellow "Compiling libuv..."
+	cd "libuv"
+	cmd /c "vcbuild.bat" "$arch" "release" "shared"
+	cp "Release/*.lib" "$libdir/"
+	cp "include/*" "$includedir/"
+	cp "Release/*.dll" "$releasedir/"
+	cp "LICENSE" "$releasedir/LICENSE.libuv"
+	cd "$oldpwd"
+}
+
+function BuildPlayd ($arch, $archdir, $build)
+{
+	Write-Yellow "Building ury-playd on $arch..."
+	$oldpwd = $pwd
+	cd "$build"
+
+	switch($arch) {
+		"x86" { $cmake_generator = "Visual Studio 14 2015"; $msbuild_platform = "Win32" }
+		"x64" { $cmake_generator = "Visual Studio 14 2015 Win64"; $msbuild_platform = "x64" }
+	}
+	Write-Yellow "Running cmake..."
+	cmake "$project" -G "$cmake_generator" -DCMAKE_PREFIX_PATH="$archdir"
+	Write-Yellow "Running msbuild..."
+	msbuild playd.sln /p:Configuration="Release" /toolsversion:14.0 /p:Platform="$msbuild_platform" /p:PlatformToolset=v140
+	cd "$oldpwd"
+}
+
+function Load-MSVC-Vars
+{
+	#Set environment variables for Visual Studio Command Prompt
+	pushd "$env:VS140COMNTOOLS"
+	cmd /c "vsvars32.bat&set" |
+	foreach {
+	  if ($_ -match "=") {
+		$v = $_.split("="); set-item -force -path "ENV:\$($v[0])"  -value "$($v[1])"
+	  }
+	}
+	popd
+	Write-Yellow "Visual Studio 2015 Command Prompt variables set."
+}
+
+# Check if an exe is Python 2.7.
+function Check-Python-Version ($pypath, $vstring)
+{
+	$python_version = & "$pypath" "--version" 2>&1
+	return $python_version -Match $vstring
+}
+
+# Check for Python 2.7, and throw an exception if not successful.
+function Check-Python ($vstring)
+{
+	if (Test-Path Env:\PYTHON)
+	{
+		if (!(Test-Path $env:PYTHON))
+		{
+			throw "Invalid path to Python '$env:PYTHON' in `$env:PYTHON."
+		}
+
+		if (Check-Python-Version $env:PYTHON $vstring)
+		{
+			return
+		}
+		else
+		{
+			throw "Invalid Python version for '$env:PYTHON' in `$env:PYTHON. We require 2.7."
+		}
+	}
+	if ((Get-Command "python.exe" -ErrorAction SilentlyContinue) -eq $null)
+	{
+		throw "Python not found in `$env:PATH or `$env:PYTHON."
+	}
+	else
+	{
+		$pypath = Get-Command "python.exe"
+		if (!(Check-Python-Version $pypath $vstring))
+		{
+			throw "Invalid Python version in `$env:PATH. We require 2.7."
+		}
+	}
+}
+
+# Main
+switch($arch) {
+	"" { $arch = "x86" }
+	"x86" {}
+	"x64" {}
+	default { throw "Invalid architecture '$arch'. Run Get-Help on this script for more info." }
+}
+
+Load-MSVC-Vars
+
+$project = "$pwd"
+$archdir = "$project\$arch"
+$depsdir = "$archdir\deps"
+$build = "$archdir\build"
+$libdir = "$archdir\lib"
+$includedir = "$archdir\include"
+$patch = "$project\patch"
+
+Write-Yellow "Making directories..."
+mkdir -Force "$depsdir"
+mkdir -Force "$build"
+mkdir -Force "$libdir"
+mkdir -Force "$includedir"
+
+if ($deps) {
+	BuildDeps $arch $depsdir $libdir $includedir $build
+}
+if ($playd) {
+	BuildPlayd $arch $archdir $build
+}
+if (!($deps -or $playd)) {
+	BuildDeps $arch $depsdir $libdir $includedir $build
+	BuildPlayd $arch $archdir $build
+}
```

Comment:
I wonder if this file (and perhaps others) in root couldn't be put in a subdir, it'd be good to keep it as clean as possible